### ISSUE TITLE
Hide NTPSI toggles in restricted regions

### ISSUE THREAD

**Issue by Miyayes** [brave/brave-browser#27717]
## Description

In https://github.com/brave/brave-browser/issues/25275, NTPSIs toggles were hidden when in restricted region. However, this seems to have regressed and they now appear again, according to @mkarolin.

- [ ] Hide NTPSI toggles when in restriction region.

**Comment by mkarolin** [comment]
Turns out it's not a regression. The SI was being hidden when Rewards isn't supported at all (by policy), but it was always visible in the restricted regions.

**Comment by kjozwiak** [comment]
The above requires `1.47.171` or higher for `1.47.x` verification 👍

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on

```
Brave | 1.47.171 Chromium: 109.0.5414.87 (Build oficial) (64 bits) 
-- | --
Revisión | 2dc18eb511c56e012081b4abc9e38c81c885f7d4-refs/branch-heads/5414@{#1241}
SO | Windows 10 Version 21H2 (Build 19044.1826)
```



Went through the STR/Cases outlined via https://github.com/brave/brave-browser/issues/27717#issue-1526535162 and ensured that the `NTP SI` toggle was removed via `Customise dashboard` -> `Show background images` as per the following:

* set the machine `language` & `region` to `Cuba`

**`1.47.169 Chromium: 109.0.5414.87 (Without Fix)`** | **`1.47.171 Chromium: 109.0.5414.87 (With Fix)`**
-----------------------------------------------------------|-------------------------------------------------------
![image](https://user-images.githubusercontent.com/38657976/212101511-0a4f1c63-b702-40e3-bc67-161ddb7faa91.png) | ![image](https://user-images.githubusercontent.com/38657976/212099371-1272c74c-f526-46ce-a21d-73b5020457a5.png)
